test/e2e/runner/start.go

```diff
@@ -41,24 +42,54 @@ func Start(ctx context.Context, logger log.Logger, testnet *e2e.Testnet) error {
 
 	// Start initial nodes (StartAt: 0)
 	logger.Info("Starting initial network nodes...")
+	wg := &sync.WaitGroup{}
+	errs := make(chan error, 2)
 	for len(nodeQueue) > 0 && nodeQueue[0].StartAt == 0 {
 		node := nodeQueue[0]
 		nodeQueue = nodeQueue[1:]
-		if err := execCompose(testnet.Dir, "up", "-d", node.Name); err != nil {
-			return err
-		}
+		wg.Add(1)
+		time.Sleep(250 * time.Millisecond) // this is to avoid breaking docker.
```

Comment:
I'm worried that this line is introducing something potentially flaky. Will Docker error if you try to invoke it too quickly? Is there some information we can get from the Docker process to determine if the tool is ready to be used again?

Reply:
legit. 

this is also via docker compose and not docker directly, so it's hard to see this happen, I get `failed to create network ci_ci: Error response from daemon: Pool overlaps with other one on this address space` as an error when running without the timeout locally. I also tried to use the `--wait` flag which doesn't appear to help.

Reply:
Yeah, `--wait` appears to just hang, which is not ideal. What if you tried running `docker-compose up --no-start` right before the loop? That should create the network so it's in place but not actually run anything yet.